### PR TITLE
add GitHub flavored markdown notes

### PR DESCRIPTION
**Pull request by swyxio**
closes https://github.com/storybooks/storybook/issues/2916#issuecomment-364205690

### CODE REVIEW

File: addons/notes/README.md [hunk 1]
```diff
@@ -3,18 +3,19 @@
 [![Build Status on CircleCI](https://circleci.com/gh/storybooks/storybook.svg?style=shield)](https://circleci.com/gh/storybooks/storybook)
 [![CodeFactor](https://www.codefactor.io/repository/github/storybooks/storybook/badge)](https://www.codefactor.io/repository/github/storybooks/storybook)
 [![Known Vulnerabilities](https://snyk.io/test/github/storybooks/storybook/8f36abfd6697e58cd76df3526b52e4b9dc894847/badge.svg)](https://snyk.io/test/github/storybooks/storybook/8f36abfd6697e58cd76df3526b52e4b9dc894847)
-[![BCH compliance](https://bettercodehub.com/edge/badge/storybooks/storybook)](https://bettercodehub.com/results/storybooks/storybook) [![codecov](https://codecov.io/gh/storybooks/storybook/branch/master/graph/badge.svg)](https://codecov.io/gh/storybooks/storybook)  
+[![BCH compliance](https://bettercodehub.com/edge/badge/storybooks/storybook)](https://bettercodehub.com/results/storybooks/storybook) [![codecov](https://codecov.io/gh/storybooks/storybook/branch/master/graph/badge.svg)](https://codecov.io/gh/storybooks/storybook)
```
Comment (sw-yx) [inline addons/notes/README.md]: looks like my Prettier reformatted this README.md a bit, this can be ignored

File: addons/notes/README.md [hunk 2]
```diff
@@ -49,25 +50,48 @@ storiesOf('Component', module)
...
+  This is some code showing usage of the component and other inline documentation
+
+  ~~~js
+  <div>
+    hello world!
+    <Component/>
+  </div>
+  ~~~
+  `)(() => <Component />)
+);
```
Comment (sw-yx) [inline addons/notes/README.md]: this is the relevant new documentation of the new exposed API. pretty simple.

File: addons/notes/README.md (hunk 1 above)
Comment (Hypnosphi) [reply addons/notes/README.md]: Can you please revert stylistic changes?

**Comment by Hypnosphi** [comment]
Please add a usage example to storybook-official

**Comment by Hypnosphi** [comment]
You probably need to run `yarn install` in project root to update lockfile

**Comment by Hypnosphi** [comment]
[Live example](https://deploy-preview-2946--storybooks-official.netlify.com/?selectedKind=Addons%7CNotes&selectedStory=withNotes%20rendering%20inline%2C%20github-flavored%20markdown&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fnotes%2Fpanel)

For some reason, it just displays raw markdown sources instead of formatted text

**Comment by Hypnosphi** [comment]
> For some reason, it just displays raw markdown sources instead of formatted text

OK, it's because of the indentation in the example. Indent is just another way to define code block

**Review by Hypnosphi — APPROVED** [review]

**Comment by sw-yx** [comment]
ahh thats why. sorry I hadn't had time last week to get around to this, glad you figured it out. was definitely working in my local setup. im still in the first month of my new job, got to figure out what my ability is to work on open source :( thank you so much for being an awesome maintainer, I hope to be a better contributor over time

**Pull request merged**